src/main/java/com/cburch/logisim/gui/start/GuiInterface.java

```diff
@@ -0,0 +1,285 @@
+
+    Toolkit.getDefaultToolkit()
+        .addAWTEventListener(this, AWTEvent.COMPONENT_EVENT_MASK | AWTEvent.CONTAINER_EVENT_MASK);
+    // pre-load the two basic component libraries, just so that the time
+    // taken is shown separately in the progress bar.
+    if (monitor != null) {
+      monitor.setProgress(SplashScreen.LIBRARIES);
+    }
+    templLoader.setParent(monitor);
+    final var count = templLoader.getBuiltin().getLibrary(BaseLibrary._ID).getTools().size()
+                    + templLoader.getBuiltin().getLibrary(GatesLibrary._ID).getTools().size();
+    assert(count >= 0);
+
+    // load in template
+    if (monitor != null) {
+      monitor.setProgress(SplashScreen.TEMPLATE_OPEN);
+    }
+    if (templateType != AppPreferences.TEMPLATE_UNKNOWN) {
+      AppPreferences.setTemplateType(templateType);
+      if (templateFile != null) AppPreferences.setTemplateFile(templateFile);
+    }
+
+    // now that the splash screen is almost gone, we do some last-minute
+    // interface initialization
+    if (monitor != null) {
+      monitor.setProgress(SplashScreen.GUI_INIT);
+    }
+    WindowManagers.initialize();
+    if (MacCompatibility.isSwingUsingScreenMenuBar()) {
+        MacCompatibility.setFramelessJMenuBar(new LogisimMenuBar(null, null, null, null));
+    } else {
+        new LogisimMenuBar(null, null, null, null);
+        // most of the time occupied here will be in loading menus, which
+        // will occur eventually anyway; we might as well do it when the
+        // monitor says we are
+    }
+
+    // Make ENTER and SPACE have the same effect for focused buttons.
+    UIManager.getDefaults()
+        .put(
+            "Button.focusInputMap",
+            new UIDefaults.LazyInputMap(
+                new Object[] {
+                    "ENTER", "pressed",
+                    "released ENTER", "released",
+                    "SPACE", "pressed",
+                    "released SPACE", "released"
+                }));
+
+    // if user has double-clicked a file to open, we'll
+    // use that as the file to open now.
+    initialized = true;
+
+    // load file
+    if (filesToOpen.isEmpty()) {
+      final var proj = ProjectActions.doNew(monitor);
+      proj.setStartupScreen(true);
+      if (monitor != null) {
+        monitor.close();
+        monitor = null;
+      }
+    } else {
+      var numOpened = 0;
+      var first = true;
+      for (final var fileToOpen : filesToOpen) {
+        try {
+          ProjectActions.doOpen(monitor, fileToOpen);
+          numOpened++;
+        } catch (LoadFailedException ex) {
+          // FIXME: report error opening: fileToOpen.getName()
+        }
+        if (first) {
+          first = false;
+          if (monitor != null) {
+            monitor.close();
+            monitor = null;
+          }
+        }
+      }
+      if (numOpened == 0) return -1;
+    }
+
+    for (final var fileToPrint : filesToPrint) {
+      doPrint(fileToPrint);
+    }
+
+    return 0;
```

Comment:
Pleasae return `RC` enum instead magic value as (IIRC) this is what this `0` stands for here.

Reply:
RC enum is NOT an exit code value. I will use a symbol in Main, as described above.